modules/options/src/main/scala/scala/build/options/ClassPathOptions.scala

```diff
@@ -7,6 +7,7 @@ import scala.build.Positioned
 final case class ClassPathOptions(
   extraRepositories: Seq[String] = Nil,
   extraClassPath: Seq[os.Path] = Nil,
+  extraSourceFiles: Seq[os.Path] = Nil,
```

Comment:
Not sure we should add that field here. Maybe it should be added in `PostBuildOptions` (even though it's more a "pre-build" thing here) or in `InternalOptions`, which are not taken into account when computing a hash of the options. But then, we have to be sure we use an `Inputs` instance that has the new inputs when computing a hash of it (in `Build.scala` IIRC).